Error when someone uses a SF screen that is not loaded

If someone else is pressing his use key while aiming at a starfall screen that is not loaded on your client, you receive the following error:

> [ERROR] addons/starfall/lua/entities/starfall_screen/cl_init.lua:76: attempt to call >method 'runScriptHook' (a nil value)
> 1. Function - addons/starfall/lua/entities/starfall_screen/cl_init.lua:76
>    1. unknown - lua/includes/modules/usermessage.lua:87

--- a/lua/entities/starfall_screen/cl_init.lua
+++ b/lua/entities/starfall_screen/cl_init.lua
@@ -68,10 +68,11 @@ net.Receive( "starfall_screen_update", function ( len )
 	end
 end )
 
+net.Receive( "starfall_screen_used", function ( len )
+	local screen = net.ReadEntity()
+	local activator = net.ReadEntity()
 
-usermessage.Hook( "starfall_screen_used", function ( data )
-	local screen = Entity( data:ReadShort() )
-	local activator = Entity( data:ReadShort() )
+	if not IsValid( screen ) then return end
 	
 	screen:runScriptHook( "starfallUsed", SF.Entities.Wrap( activator ) )
 	

--- a/lua/entities/starfall_screen/init.lua
+++ b/lua/entities/starfall_screen/init.lua
@@ -8,8 +8,9 @@ assert(SF, "Starfall didn't load correctly!")
 local context = SF.CreateContext()
 local screens = {}
 
-util.AddNetworkString("starfall_screen_download")
-util.AddNetworkString("starfall_screen_update")
+util.AddNetworkString( "starfall_screen_download" )
+util.AddNetworkString( "starfall_screen_update" )
+util.AddNetworkString( "starfall_screen_used" )
 
 local function sendScreenCode ( screen, owner, files, mainfile, recipient )
 	net.Start( "starfall_screen_download" )
@@ -154,13 +155,13 @@ function ENT:Think()
 	return true
 end
 
--- Sends a umsg to all clients about the use.
+-- Sends a net message to all clients about the use.
 function ENT:Use( activator )
 	if activator:IsPlayer() then
-		umsg.Start( "starfall_screen_used" )
-			umsg.Short( self:EntIndex() )
-			umsg.Short( activator:EntIndex() )
-		umsg.End( )
+		net.Start( "starfall_screen_used" )
+			net.WriteEntity( self )
+			net.WriteEntity( activator )
+		net.Broadcast()
 	end
 	if self.sharedscreen then
 		self:runScriptHook( "starfallUsed", SF.Entities.Wrap( activator ) )

